### PR TITLE
ESLint 6 Update

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Testing out [`eslint-config-terra` update](https://github.com/cerner/eslint-config-terra/pull/34)

Changes include:
* Object.assign usage switched to Object spread syntax: https://eslint.org/docs/rules/prefer-object-spread#prefer-use-of-an-object-spread-over-objectassign-prefer-object-spread
* Formatting for curly braces and new lines
```diff
-  )
-}
+)}
```
* Disable compat linter for `maxTouchPoints`. We don't polyfill this but we check for browser specific equivalents in browsers that don't support `maxTouchPoints`, e.g. `msMaxTouchPoints`. We  are also slated to remove the `maxTouchPoints` code when we refactor Checkbox and Radio button.

### CODE REVIEW

File: packages/terra-hookshot/tests/jest/HookshotDefaultExample.jsx [hunk 1]
```diff
@@ -35,7 +35,13 @@ class HookshotDefaultExample extends React.Component {
         >
           {hookshotContent}
         </Hookshot>
-        <button type="button" id="hookshot-standard-button" onClick={this.handleButtonClick} />
+        <button
+          type="button"
+          id="hookshot-standard-button"
+          onClick={this.handleButtonClick}
+        >
+          {''}
```
Comment (mjhenkes) [inline packages/terra-hookshot/tests/jest/HookshotDefaultExample.jsx]: why the empty space?

Comment (mjhenkes) [reply packages/terra-hookshot/tests/jest/HookshotDefaultExample.jsx]: Would this work?
```jsx
<button
  type="button"
  id="hookshot-standard-button"
  onClick={this.handleButtonClick}
/>
```

Comment (mjhenkes) [reply packages/terra-hookshot/tests/jest/HookshotDefaultExample.jsx]: I have the distinct feeling this was probably talked about during scrum and i wasn't paying attention.

Comment (bjankord) [reply packages/terra-hookshot/tests/jest/HookshotDefaultExample.jsx]: Making eslint-jsx-a11y plugin happy without updating wdio screenshots. The eslint-jsx-a11y plugin expects buttons to have text. In this hookshot test, we have a button with no text. Using `{''}` for the button text by-passes eslint-jsx-a11y plugin as it is not able to parse the code within `{}` and also keeps the wdio screenshot the same.

Options I can see are we can disable the lint rule here, add text to the button and regen wdio screenshots, or go with this as-is.

Comment (mjhenkes) [reply packages/terra-hookshot/tests/jest/HookshotDefaultExample.jsx]: good enough for me, just wondering why

**Pull request merged**